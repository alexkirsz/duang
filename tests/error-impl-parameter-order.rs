// this test check the error message when mandatory argument come after default ones.
use duang::duang;

duang!(
pub fn foo<T>(a: T,b: f64 = 13.0, c: T) -> (T,f64,T)
where
  T: std::ops::Mul<T, Output = T>,
  T: std::fmt::Display,
  T: Copy,
{
  (a,b,c)
}
);

fn main() {
}
