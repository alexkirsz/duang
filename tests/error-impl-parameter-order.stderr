error: Mandatory parameter must be ahead of default parameter.
 --> $DIR/error-impl-parameter-order.rs:5:35
  |
5 | pub fn foo<T>(a: T,b: f64 = 13.0, c: T) -> (T,f64,T)
  |                                   ^
